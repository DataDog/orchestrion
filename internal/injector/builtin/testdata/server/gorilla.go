// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"io"
	"net/http"

	"github.com/gorilla/mux"
//line <generated>:1
	"github.com/datadog/orchestrion/instrument"
	mux1 "gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux"
)

//line samples/server/gorilla.go:15
func gorillaMuxServer() {
	r :=
//line <generated>:1
		mux1.WrapRouter(
//line samples/server/gorilla.go:16
			mux.NewRouter())
//line samples/server/gorilla.go:17
	ping :=
//line <generated>:1
		instrument.WrapHandlerFunc(
//line samples/server/gorilla.go:17
			func(w http.ResponseWriter, r *http.Request) {
				w.Header().Set("Content-Type", "application/json")
				w.WriteHeader(http.StatusOK)

				_, _ = io.WriteString(w, `{"message":"pong"}`)
			})
//line samples/server/gorilla.go:23
	r.HandleFunc("/ping", ping).Methods("GET")
	_ = http.ListenAndServe(":8080", r)
}
