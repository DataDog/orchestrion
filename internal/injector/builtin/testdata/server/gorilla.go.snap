// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"io"
	"net/http"

	"github.com/gorilla/mux"
//line <generated>:1
	__orchestrion_instrument "github.com/datadog/orchestrion/instrument"
)

//line samples/server/gorilla.go:15
func gorillaMuxServer() {
	r := mux.NewRouter()
	ping :=
//line <generated>:1
		__orchestrion_instrument.WrapHandlerFunc(
//line samples/server/gorilla.go:17
			func(w http.ResponseWriter, r *http.Request) {
				w.Header().Set("Content-Type", "application/json")
				w.WriteHeader(http.StatusOK)

				_, _ = io.WriteString(w, `{"message":"pong"}`)
			})
//line samples/server/gorilla.go:23
	r.HandleFunc("/ping", ping).Methods("GET")
	_ = http.ListenAndServe(":8080", r)
}
